src/field_theory/polynomial_galois_group.lean

```diff
@@ -0,0 +1,245 @@
+/-
+Copyright (c) 2020 Thomas Browning and Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning and Patrick Lutz
+-/
+
+import field_theory.galois
+
+/-!
+# Galois Groups of Polynomials
+
+In this file we introduce the Galois group of a polynomial, defined as
+the automorphism group of the splitting field.
+
+## Main definitions
+
+- `gal p`: the Galois group of a polynomial p.
+- `restrict p E`: the restriction homomorphism `(E ≃ₐ[F] E) → gal p`.
+- `gal_action p E`: the action of `gal p` on the roots of `p` in `E`.
+
+## Main results
+
+- `restrict_smul`: `restrict p E` is compatible with `gal_action p E`.
+- `gal_action_hom_injective`: the action of `gal p` on the roots of `p` in `E` is faithful.
+- `restrict_prod_inj`: `gal (p * q)` embeds as a subgroup of `gal p × gal q`.
+-/
+
+noncomputable theory
+open_locale classical
+
+open finite_dimensional
+
+namespace polynomial
+
+variables {F : Type*} [field F] (p q : polynomial F) (E : Type*) [field E] [algebra F E]
+
+/-- The Galois group of a polynomial -/
+@[reducible] def gal := p.splitting_field ≃ₐ[F] p.splitting_field
```

Comment:
I'm not sure this should be reducible.
Rather, I would add something like `@[derive [group, fintype]]` or something like that.

But maybe lots of things break if you make this semireducible. It's certainly possible to convince me to keep the reducibility (-;

Reply:
Nice! Seems like that works (I had to add `has_coe_to_fun`)